types/shared.d.ts

```diff
@@ -374,13 +374,37 @@ export interface IStripeConnectInitParams {
   locale?: string;
 }
 
+type ConnectHTMLElementRecordBase = {
+  [tagName in ConnectElementTagName]: HTMLElement;
```

Comment:
what happens for internal components like debug-utils?

Reply:
We can use their custom methods, but there won't be auto-completion from typescript

Reply:
changed it to return `HTMLElement | null`; returning `HTMLElement` seems a bit odd to me since it doesn't really returns `HTMLElement` for external users, but LMK what y'all think

Reply:
@kaiying-stripe `| null` doesn't make sense to me, but that is not how you have it now right? I pulled the branch and things work as I expected them to (does not return null, has the additional setter methods)

Reply:
Ah no, I meant there's `| null` when the value is not in `ConnectElementTagName`

Reply:
so the hovered variable in the screenshot would have type `HTMLElement | null`

Reply:
Ack, ideally that would be `HtmlElement` without the `| null`, however that isn't really a big problem as this should only happen when someone is doing some intentional casting